### PR TITLE
Revalidate some transactions on every block import.

### PR DESCRIPTION
**Pull request by tomusdrw**
This should fix transactions stalling in the pool for a long time.

I'm still running tests for Kusama to make sure it works correctly.

### CODE REVIEW

File: client/transaction-pool/src/maintainer.rs [hunk 1]
```diff
@@ -74,6 +74,14 @@ where
 		id: &BlockId<Block>,
 		retracted: &[Block::Hash],
 	) -> Box<dyn Future<Output=()> + Send + Unpin> {
+		let now = std::time::Instant::now();
+		let took = move || {
+			let took = std::time::Instant::now().duration_since(now);
+			format!("Took {} ms", took.as_millis())
+		};
+
```
Comment (marcio-diaz) [inline client/transaction-pool/src/maintainer.rs]: Next time you can do something like
```
use tracing;
let span = tracing::span!(tracing::Level::DEBUG, "retracted");
let _enter = span.enter();
```
inside a context, to send the time to trace/grafana.

Comment (tomusdrw) [reply client/transaction-pool/src/maintainer.rs]: Can I also display that in logs?

Comment (marcio-diaz) [reply client/transaction-pool/src/maintainer.rs]: Yeah, you need to use `--tracing-targets`  with an appropriated pattern (maybe `substrate_client`) and `--tracing-receiver` (but is defaulting to Log).

**Comment by tomusdrw** [comment]
Run some tests on Kusama and it seems to be working correctly (as in: not breaking anything). We now revalidate 16 transactions from the front of the transaction pool (after pruning), to make sure that what we have is still valid for the next block.

**Comment by gavofyork** [comment]
Merged since CI failure (no connectivity) clearly spurious.

**Pull request merged**